src/print.jl

```diff
@@ -634,13 +651,32 @@ function constraint_string(
         return prefix * constraint_str
     end
 end
+
+"""
+    constraint_string(
+        print_mode,
+        ref::ConstraintRef;
+        in_math_mode::Bool = false)
+
+Return a string representation of the constraint `ref`, given the `print_mode`.
+
+`print_mode` should be `IJuliaMode` or `REPLMode`.
+"""
 function constraint_string(print_mode, ref::ConstraintRef; in_math_mode = false)
```

Comment:
Should these even be public methods? There are `Base.show(io::IO, ref::ConstraintRef)` and `Base.show(io::IO, ::MIME"text/latex", ref::ConstraintRef)`, which we should encourage the use-of instead.

Reply:
Yes, extension might add methods to this function

Reply:
The could implement `show` directly though, and we could use that.

Reply:
I think the difference is that `is_math_mode` can be used to avoid `$$` to be added twice.